### PR TITLE
New routing with new vue memory management

### PR DESCRIPTION
**Pull request by PaulMansat**
- solve the issue that resulted in a task's component being mounted multiple times (in fact, to see this issue, one had to first log into one task and then move to another task, there the user would be connected twice to the server because the component would be mounted twice)
- to solve the issue a new routing procedure is introduced with a new architecture: all components share a global main component where the routes for the child components (i.e desc. component, training component, and testing component are called). I will add in the documentation a scheme of the new routing procedure.
- the global task component being shared amongst all tasks, it has to store its state in a Vuex store (used to store custom component's state)
- Vuex is added as a new dependency (to work the npm dependencies of the application need to update by running the command npm install)
- this issue was also related to the fact that some variables used in a component were defined outside the component's scope. This meant that these variables were shared by all of the component's instances. Originally, these variables were used to store a TFJS model, because TFJS models cannot be defined in the component's data, due to memory conflicts between TensforflowJS and VueJS. The solution is to store the TFJS model in indexedDB, and to load the model only when we need it (i.e only when training and not before in a component).
- the new routing procedure required some modifications in many files, that is the reason why this pull request is quite large.
- finally, a minor correction was added to the communication manager. The port number in the communication number was fixed to 9000 and was not connected to the port number information defined in the task's training information.

### CODE REVIEW

File: mobile-browser-based-version/src/components/building_frames/ImageTestingFrame.vue [hunk 1]
```diff
@@ -326,17 +324,12 @@ export default {
     },
     downloadPredictionsCsv(){
       console.log(this.predictions)
-      let pred = ""
-      let header_length = 0
-      for (const [id, prediction] of Object.entries(this.predictions)){
-        header_length = prediction.length
-        pred += id + ","+ prediction.map(dict => dict["className"]+","+dict["probability"]).join(',')+"\n"
```
Comment (PaulMansat) [inline mobile-browser-based-version/src/components/building_frames/ImageTestingFrame.vue:333]: I might have deleted a part you've updated @MarcelTorne. Could you confirm that the new version is correct? Thanks!

File: mobile-browser-based-version/src/helpers/communication_script/helpers.js [hunk 2]
```diff
@@ -44,19 +44,13 @@ export function assignWeightsToModel(serializedWeights, model) {
...
-            (serializedWeights, peer) =>{
-                const serializedWeight = serializedWeights[idx]["$variable"];
-                const tensor = deserializeTensor(serializedWeight.val);
-                tensorSum = tensor.add(tensorSum)
-                tensor.dispose()
-            }
-        )
-        weight.val.assign(tensorSum.div(peersSerializedWeights.length + 1)); //average
-        tensorSum.dispose();
+        const serializedWeight = serializedWeights[idx]["$variable"];
```
Comment (PaulMansat) [inline mobile-browser-based-version/src/helpers/communication_script/helpers.js:49]: I might have overwritten some files again...

Comment (PaulMansat) [reply mobile-browser-based-version/src/helpers/communication_script/helpers.js]: Resolved: brought back the old file

File: mobile-browser-based-version/src/components/building_frames/ImageTestingFrame.vue (hunk 1 above)
Comment (PaulMansat) [reply mobile-browser-based-version/src/components/building_frames/ImageTestingFrame.vue]: Resolved: brought back the original file in from the maste branch.

File: mobile-browser-based-version/src/task_definition/mnist.js [hunk 3]
```diff
@@ -201,14 +200,12 @@ export class MnistTask {
...
-                classes_dict[i] = classes
-                
-                i++
+                classes_array.push(classes)
             }
 
             console.log("Prediction Sucessful!")
 
-            return classes_dict;
+            return classes_array;
```
Comment (PaulMansat) [inline mobile-browser-based-version/src/task_definition/mnist.js]: Resolved: brought bach the original file from the master branch.

File: mobile-browser-based-version/src/task_definition/lus_covid.js [hunk 4]
```diff
@@ -197,72 +191,6 @@ export class LusCovidTask {
...
-        }
-
-        const xs = tf.concat(xsArray, 0)
-    
-        console.log(xs)
-        console.log(patients)
-
-        return {xTest: xs, ids: patients}
-    }
-
```
Comment (PaulMansat) [inline mobile-browser-based-version/src/task_definition/lus_covid.js]: Resolved: brought back the original branch from the master branch.

**Review by martinjaggi — DISMISSED** [review]

File: mobile-browser-based-version/src/store/store.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import {createStore} from 'vuex'
```
Comment (mmilenkoski) [inline mobile-browser-based-version/src/store/store.js:1]: this file needs to be updated as we discussed

**Review by mmilenkoski — CHANGES_REQUESTED** [review]

File: mobile-browser-based-version/src/store/store.js (hunk 5 above)
Comment (PaulMansat) [reply mobile-browser-based-version/src/store/store.js:1]: The issue is resolved, the new code is in the file MainTaskFrame and not GlobalTaskFrame 👍

**Review by mmilenkoski — APPROVED** [review]
Thanks a lot for working on this Paul!

**Review by mmilenkoski — APPROVED** [review]

**Pull request merged**